x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/BetweenFunctionProcessor.java

```diff
@@ -0,0 +1,134 @@
+
+    public static final String NAME = "sbtw";
+
+    private final Processor source, left, right, greedy, caseSensitive;
+
+    public BetweenFunctionProcessor(Processor source, Processor left, Processor right, Processor greedy, Processor caseSensitive) {
+        this.source = source;
+        this.left = left;
+        this.right = right;
+        this.greedy = greedy;
+        this.caseSensitive = caseSensitive;
+    }
+
+    public BetweenFunctionProcessor(StreamInput in) throws IOException {
+        source = in.readNamedWriteable(Processor.class);
+        left = in.readNamedWriteable(Processor.class);
+        right = in.readNamedWriteable(Processor.class);
+        greedy = in.readNamedWriteable(Processor.class);
+        caseSensitive = in.readNamedWriteable(Processor.class);
+    }
+
+    @Override
+    public final void writeTo(StreamOutput out) throws IOException {
+        out.writeNamedWriteable(source);
+        out.writeNamedWriteable(left);
+        out.writeNamedWriteable(right);
+        out.writeNamedWriteable(greedy);
+        out.writeNamedWriteable(caseSensitive);
+    }
+
+    @Override
+    public Object process(Object input) {
+        return doProcess(source.process(input), left.process(input), right.process(input),
+                greedy.process(input), caseSensitive.process(input));
+    }
+
+    public static Object doProcess(Object source, Object left, Object right, Object greedy, Object caseSensitive) {
+        if (source == null) {
+            return null;
+        }
+
+        throwIfNotString(source);
+        throwIfNotString(left);
+        throwIfNotString(right);
+
+        throwIfNotBoolean(greedy);
+        throwIfNotBoolean(caseSensitive);
+
+        String str = source.toString();
+        String strRight = right.toString();
+        String strLeft = left.toString();
+        boolean bGreedy = ((Boolean) greedy).booleanValue();
+        boolean bCaseSensitive = ((Boolean) caseSensitive).booleanValue();
+        return StringUtils.between(str, strLeft, strRight, bGreedy, bCaseSensitive);
+    }
+
+    private static void throwIfNotString(Object obj) {
+        if (!(obj instanceof String || obj instanceof Character)) {
+            throw new EqlIllegalArgumentException("A string/char is required; received [{}]", obj);
+        }
+    }
+
+    private static void throwIfNotBoolean(Object obj) {
+        if (!(obj instanceof Boolean)) {
+            throw new EqlIllegalArgumentException("A boolean is required; received [{}]", obj);
+        }
+    }
+
+    protected Processor source() {
+        return source;
+    }
+
+    public Processor left() {
+        return left;
+    }
+
+    public Processor right() {
+        return right;
+    }
+
+    public Processor greedy() {
+        return greedy;
+    }
+
+    public Processor caseSensitive() {
+        return caseSensitive;
+    }
+
+
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
Small nit - the getNameWritable tends to be towards the top after `writeTo` and `hashCode` before `equals`.